infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NamespaceProvisioner.java

```diff
@@ -11,167 +11,64 @@
-import io.fabric8.kubernetes.api.model.Secret;
-import io.fabric8.kubernetes.api.model.SecretBuilder;
-import io.fabric8.kubernetes.client.KubernetesClientException;
-import java.util.Base64;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
+import io.fabric8.kubernetes.api.model.Namespace;
 import javax.inject.Inject;
-import org.eclipse.che.api.core.NotFoundException;
-import org.eclipse.che.api.core.ServerException;
-import org.eclipse.che.api.core.model.user.User;
-import org.eclipse.che.api.user.server.PreferenceManager;
-import org.eclipse.che.api.user.server.UserManager;
+import org.eclipse.che.api.workspace.server.model.impl.RuntimeIdentityImpl;
 import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
 import org.eclipse.che.api.workspace.server.spi.NamespaceResolutionContext;
-import org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesClientFactory;
 import org.eclipse.che.workspace.infrastructure.kubernetes.api.shared.KubernetesNamespaceMeta;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespace;
 import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.configurator.NamespaceConfigurator;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.configurator.UserPreferencesConfigurator;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.configurator.UserProfileConfigurator;
 
 /**
- * On namespace provisioning, creates k8s {@link Secret} profile and preferences from information
- * about the User.
+ * Provisions the k8s {@link Namespace}. After provisioning, configures the namespace through {@link
+ * NamespaceConfigurator}.
  *
  * @author Pavol Baran
  */
 public class NamespaceProvisioner {
-  private static final Logger LOG = LoggerFactory.getLogger(NamespaceProvisioner.class);
-  private static final String USER_PROFILE_SECRET_NAME = "user-profile";
-  private static final String USER_PREFERENCES_SECRET_NAME = "user-preferences";
-  private static final String USER_PROFILE_SECRET_MOUNT_PATH = "/config/user/profile";
-  private static final String USER_PREFERENCES_SECRET_MOUNT_PATH = "/config/user/preferences";
-
   private final KubernetesNamespaceFactory namespaceFactory;
-  private final KubernetesClientFactory clientFactory;
-  private final UserManager userManager;
-  private final PreferenceManager preferenceManager;
+  private final UserProfileConfigurator userProfileConfigurator;
+  private final UserPreferencesConfigurator userPreferencesConfigurator;
 
   @Inject
   public NamespaceProvisioner(
       KubernetesNamespaceFactory namespaceFactory,
-      KubernetesClientFactory clientFactory,
-      UserManager userManager,
-      PreferenceManager preferenceManager) {
+      UserProfileConfigurator userProfileConfigurator,
```

Comment:
you can inject Set<Interface> with multibinder.

Reply:
yes, thank you